cmd/pocket_core/main.go

```diff
@@ -48,20 +49,33 @@ func chainsFromFile(){
 	}
 }
 
+//NOTE: this is for centralized dispatch and may be removed at production
+func sendEntryMessage(){
+	m:=message.NewEnterNetMessage()
+	message.SendMessage(message.RELAY, m, _const.DISPATCHIP, message.EnterNetworkPayload{})
+}
+
+func sendExitMessage(){
+	m:=message.NewExitNetMessage()
+	message.SendMessage(message.RELAY, m, _const.DISPATCHIP, message.ExitNetworkPayload{})
+}
+
 /*
 "startClient" Starts the client with the given initial configuration.
 */
 func startClient() {
-	config.InitializeConfiguration()                                  // initializes the configuration from flags and defaults.
-	config.BuildConfiguration()                                       // builds the proper structure on pc for core client to operate.
-	config.PrintConfiguration()                                       // print the configuration the the cmd.
-	peersFromFile()                                                   // check for manual peers
-	node.GetPeerList().AddPeersToDispatchStructure()                  // add peers to dispatch structure
-	chainsFromFile()                                                  // check for chains.json file
-	logs.NewLog("Started client", logs.InfoLevel, logs.JSONLogFormat) // log start message
+	config.InitializeConfiguration()                                			// initializes the configuration from flags and defaults.
+	config.BuildConfiguration()                                     			// builds the proper structure on pc for core client to operate.
+	config.PrintConfiguration()                                     			// print the configuration the the cmd.
+	peersFromFile()                                                   			// check for manual peers
+	node.GetPeerList().AddPeersToDispatchStructure()							// add peers to dispatch structure
+	chainsFromFile()															// check for chains.json file
+	logs.NewLog("Started client", logs.InfoLevel, logs.JSONLogFormat) 	// log start message
 	rpc.StartAPIServers()                                             // runs the server endpoints for client and relay api.
-	message.RunMessageServers()                                       // runs servers for messages
-	fmt.Print("Press any key + 'Return' to quit: ")                   // prompt user to exit
-	input := bufio.NewScanner(os.Stdin)                               // unnecessary temporary entry
-	input.Scan()                                                      // wait
+	message.RunMessageServers()													// runs servers for messages
+	sendEntryMessage()															// send entry message
+	fmt.Print("Press any key + 'Return' to quit: ")                 			// prompt user to exit to dispatcher
+	input := bufio.NewScanner(os.Stdin)                             			// unnecessary temporary entry
+	input.Scan()                                                    			// wait
+	sendExitMessage()															// send exit message to dispatcher
```

Comment:
Maybe this should be "deferred", so that if we exit in another way, we still get the message to go through most times.

Reply:
Nice! Good call. However, I believe OS.Exit() does not execute deffered functions, wondering if there is a better way to exit the program gracefully?